_posts/2020-08-31-rico-logging.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title:  'Rico logging'
+author: TODO
+featuredImage: TODO
+excerpt: 'TODO'
+permalink: '/rico/2020/08/29/rico-2.0-logging.html'
+categories: [Java, Jakarta EE, Java EE, Spring, Rico, Logging, GELF]
+header:
+  text: Rico Logging
+  image: post
+---
+
+Back in the days when we had one server and maybe ten concurrent users.
+When we used to shut down the system for half a day to deploy a new release.
+When browsers version numbers were single digits.
+Back then we did read log files like books.
+I mean not for leisure, no no!
+I want to say if a customer called and reported a problem we used to do the following:
+We grabbed a fresh cup of coffee and cracked our fingers.
+Maybe we called the intern over to watch and learn.
+Then we sshed into the production server.
+Copied the log file to our local machine and started to grep for error messages.
+Once we found the interesting spot we read through the log line by line to understand what had happened.
+We were like Cypher in "Matrix" seeing through the code.
+
+Like some other things this has changed since 1999 when Matrix was released.
+Today there are many servers, DB, load-balancer, reverse proxy and much more generating logs.
+Instances come and go as user numbers increase and decrease.
+Hopefully there are more than ten users online at the same time.
+Also there is better no way for any developer to ssh into production servers anymore.
+
+What has not changed is that logs give us valuable information on what has happened on our system.
+Its just that this information is no longer generated by a single place.
+Rather it is fragmented and spread over many servers.
+Some may be offline at the moment
+To still be able to have a complete picture about today's polyglot systems all the logs need to be collected in a central location.
+
```

Comment:
I would add a headline here ("Centralized Logging") and move the picture 3 lines down.